library/certificate_request.py

```diff
@@ -108,62 +113,64 @@
       - cRLSign
       - encipherOnly
       - decipherOnly
-    required: false
+    type: list
+    elements: str
     default:
       - digitalSignature
       - keyEncipherment
   extended_key_usage:
     description:
       - Extended Key Usage attributes to be present in the
         certificate request.
-    required: false
     default:
       - id-kp-serverAuth
       - id-kp-clientAuth
+    type: list
+    elements: str
   auto_renew:
     description:
       - Indicates if the certificate should be renewed
         automatically before it expires.
-    required: false
+    type: bool
     default: true
   wait:
     description:
       - If the role should block while waiting for the certificate
         to be issued.
-    required: false
+    type: bool
     default: true
   country:
     description:
       - Country requested for the certificate subject.
-    required: false
+    type: str
   state:
     description:
       - State requested for the certificate subject.
-    required: false
+    type: str
   locality:
     description:
       - Locality requested for the certificate subject (usually city).
-    required: false
+    type: str
   organization:
     description:
       - Organization requested for the certificate subject.
-    required: false
+    type: str
   organizational_unit:
     description:
       - Organizational unit requested for the certificate subject.
-    required: false
+    type: str
   contact_email:
     description:
       - Contact email requested for the certificate subject.
-    required: false
+    type: str
   run_before:
     description:
       - Command that should run before saving the certificate.
-    required: false
+    type: str
   run_after:
     description:
       - Command that should run after saving the certificate.
-    required: false
```

Comment:
Why are you removing `required: false`?

Reply:
The default is `required: false` - better style to omit `required` unless `required: true`